### DISCUSSION TITLE
[Bug] Checkout Saferpay: Credit card payment failed

### DISCUSSION THREAD

**Discussion by josias-hosang** [swellstores/community#415] [Help] [answered]
Ahoi! This is really severe bug! I tested in a live environment with actual credit card payment. The payment was successfully and leading me back to the checkout page, stating: `Payment method not supported by this gateway.` (see screenshot). Therefore the order was NOT completed but my credit card was balanced. We are using the Saferpay plugin and I tried MasterCard. With TWINT over the same gateway it's working and using the test environment of Saferpay was also successful. But this is a problem leaving me hanging with no logs or possibilities to fix this. Thanks for helping me soon.

![Screenshot 2023-10-24 at 15-42-34 Checkout](https://github.com/swellstores/community/assets/48205514/e4ce0e8d-646e-4ddb-a584-ab6ef9dd2bbd)

**Comment by steven-k1** [comment]
@josias-hosang Thanks for the details. Looking into this.

**Reply by logeshswell** [reply]
@josias-hosang When I checked the logs, I saw that the transaction was canceled and Saferpay returned the below response for your payment:
```json
{
  "result": {
    "token": "jfdp6tuyne9cj2l5riingz9tk",
    "redirect_url": "https://www.saferpay.com/vt2/api/PaymentPage/340831/17753556/jfdp6tuyne9cj2l5riingz9tk"
  }
}
```

The redirect URL returns back to the checkout page as follows:

<img width="1512" alt="Screenshot 2023-10-25 at 12 16 29 PM" src="https://github.com/swellstores/community/assets/100766131/7019cd82-7d99-45f7-8abf-8aca59ab6587">

But, you were saying that it has thrown a different error message which is confusing now. So, can you retry another order with a screen recording, so that it would help us confirm if it's an issue?

On a side note, I also verified that live payments are being processed successfully via Saferpay with other merchants.

**Reply by logeshswell** [reply]
Secondly, since the transaction is marked as canceled by Saferpay, you will receive a refund from the gateway if the payment is deducted from your card. Please check the refund status & timeline with the Saferpay team by sharing the `token` value as the payment reference.

**Reply by josias-hosang** [reply]
Ahoi @logeshswell Thanks for investigating this. Of course I cancelled the payment manually in Saferpay. Now, I placed a new order and leave the payment open for a moment so you can check: https://merz.swell.store/checkout/20d7b89258cf356a6b8e920a74ef54bb/payment?gateway=saferpay&redirect_status=succeeded

**Reply by logeshswell** [reply, marked as answer]
Thanks for helping me reproduce the issue.
As I can check the logs, Saferpay returned the following (masked) response as part of the redirection.
```json
{
  "Transaction": {
    "Type": "PAYMENT",
    "Status": "AUTHORIZED"
    ...
  },
  "RegistrationResult": {
    "Success": false,
    "Error": {
      "ErrorName": "PERMISSION_DENIED",
      "ErrorMessage": "Secure Card Data Service access denied for Customer 3****31."
    }
  },
  ...
}
```

Can you check with the Saferpay team to enable **Secure Card Data Service** for your account?
After it is enabled, it would work fine as expected.

**Reply by josias-hosang** [reply]
Oh, ok. This service is to save CC information to reuse it. Probably it is at an additional cost (I will check this); our customer will not use this feature and so not willing to pay more. Is it also possible to disable saving of CC information in swell store? That would be nice.

**Reply by logeshswell** [reply]
> Is it also possible to disable saving of CC information in swell store?

No, I don't think it is possible at the moment. But, you can request to enable the feature with Saferpay team.
